src/server/enrollment-system.js

```diff
@@ -932,21 +932,11 @@ function veteranToFinancialsInfo(veteran) {
   });
 
   const dependentFinancials = veteranToDependentFinancialsCollection(veteran);
-  let hasDependentFinancials = false;
-  if (dependentFinancials) {
-    hasDependentFinancials = _.compact(dependentFinancials.dependentFinancials.map((child) => { return child.incomes; })).length > 0;
-  }
   const spouseFinancials = veteranToSpouseFinancials(veteran);
-  const hasSpouseIncome = spouseFinancials && spouseFinancials.spouseFinancials.incomes;
-
-  const hasIncomeData = expenses || incomes || hasSpouseIncome || hasDependentFinancials;
-
-  if (!hasIncomeData) {
-    return undefined;
-  }
+  const discloseFinancials = veteran.discloseFinancialInformation;
 
```

Comment:
Based on my reading of the PR description, I think we want to just do something like
```
if (!veteran.discloseFinancialInformation) { 
    return undefined;
}
```
here so that `marriedLastCalendarYear` and `numberOfDependentChildren` aren't forcing `financialStatement` to be present if `discloseFinancialInformation` is false.